anta/result_manager/models.py

```diff
@@ -102,6 +81,34 @@ def is_error(self, message: str | None = None) -> None:
         """
         self._set_status(AntaTestStatus.ERROR, message)
 
+
+class AtomicTestResult(BaseTestResult):
+    """Describe the result of an atomic test part of a larger test related to a TestResult instance.
+
+    Attributes
+    ----------
+    description : str | None
+        Description of the AtomicTestResult.
+    inputs: BaseModel | None
+        If this AtomicTestResult is related to a specific parent test input, this field must be set.
+    result : AntaTestStatus
+        Result of the atomic test.
+    messages : list[str]
+        Messages reported by the test.
+    """
+
+    _parent: TestResult
+    description: str
+    inputs: SerializeAsAny[InstanceOf[BaseModel]] | None = None
```

Comment:
could it not be `AntaTest.Inputs`?

Reply:
Yes it should be. But it causes circular import.
See https://github.com/aristanetworks/anta/issues/994, to implement in another PR.